pkg/kubelet/kubelet.go

```diff
@@ -699,11 +699,19 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		}
 		// TODO(roberthbailey): Can we do this without having credentials to talk
 		// to the cloud provider?
-		instanceID, err := instances.ExternalID(kl.hostname)
+		externalID, err := instances.ExternalID(kl.hostname)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get external ID from cloud provider: %v", err)
+		}
+		node.Spec.ExternalID = externalID
+		// TODO: We can't assume that the node has credentials to talk to the
+		// cloudprovider from arbitrary nodes. At most, we should talk to a
+		// local metadata server here.
```

Comment:
It shouldn't be difficult to use the local metadata server in the GCE implementation of this method (since it's only ever used from the local node). If you don't want to fix it as part of this PR, feel free to create an issue and assign it to me.